doc/ApiDocGeneration/Generate-DocIndex.ps1

```diff
@@ -1,152 +1,150 @@
+    #& $($DocFx) init -q -o "${DocOutDir}"
+    # The line below is used for testing in local
+    docfx init -q -o "${DocOutDir}"
+    LogDebug "Copying template and configuration..."
+    New-Item -Path "${DocOutDir}" -Name "templates" -ItemType "directory" -Force
+    Copy-Item "${DocGenDir}/templates/*" -Destination "${DocOutDir}/templates" -Force -Recurse
+    Copy-Item "${DocGenDir}/docfx.json" -Destination "${DocOutDir}/" -Force
+    $YmlPath = "${DocOutDir}/api"
+    New-Item -Path $YmlPath -Name "toc.yml" -Force
+    $visitedService = @{}
+    # Sort and display toc service name by alphabetical order, and then sort artifact by order.
+    foreach ($serviceMapping in ($tocContent.GetEnumerator() | Sort-Object Value, Key)) {
+        $artifact = $serviceMapping.Key
+        $serviceName = $serviceMapping.Value
+        $fileName = ($serviceName -replace '\s', '').ToLower().Trim()
+        if ($visitedService.ContainsKey($serviceName)) {
+            Add-Content -Path "$($YmlPath)/${fileName}.md" -Value "#### $artifact"
+        }
+        else {
+            Add-Content -Path "$($YmlPath)/toc.yml" -Value "- name: ${serviceName}`r`n  href: ${fileName}.md"
+            New-Item -Path $YmlPath -Name "${fileName}.md" -Force
+            Add-Content -Path "$($YmlPath)/${fileName}.md" -Value "#### $artifact"
+            $visitedService[$serviceName] = $true
         }
     }
-}
 
-Write-Verbose "Creating Site Title and Navigation..."
-New-Item -Path "${DocOutDir}" -Name "toc.yml" -Force
-Add-Content -Path "${DocOutDir}/toc.yml" -Value "- name: Azure SDK for NET APIs`r`n  href: api/`r`n  homepage: api/index.md"
+    # Generate toc homepage.
+    LogDebug "Creating Site Title and Navigation..."
+    New-Item -Path "${DocOutDir}" -Name "toc.yml" -Force
+    Add-Content -Path "${DocOutDir}/toc.yml" -Value "- name: Azure SDK for $lang APIs`r`n  href: api/`r`n  homepage: api/index.md"
 
-Write-Verbose "Copying root markdowns"
-Copy-Item "$($RepoRoot)/README.md" -Destination "${DocOutDir}/api/index.md" -Force
-Copy-Item "$($RepoRoot)/CONTRIBUTING.md" -Destination "${DocOutDir}/api/CONTRIBUTING.md" -Force
+    LogDebug "Copying root markdowns"
+    Copy-Item "$($RepoRoot)/README.md" -Destination "${DocOutDir}/api/index.md" -Force
+    Copy-Item "$($RepoRoot)/CONTRIBUTING.md" -Destination "${DocOutDir}/api/CONTRIBUTING.md" -Force
 
-Write-Verbose "Building site..."
-& "${DocFxTool}" build "${DocOutDir}/docfx.json"
+    LogDebug "Building site..."
+    #& $($DocFx) build "${DocOutDir}/docfx.json"
+    # The line below is used for testing in local
+    docfx build "${DocOutDir}/docfx.json"
+    Copy-Item "${DocGenDir}/assets/logo.svg" -Destination "${DocOutDir}/_site/" -Force    
+}
 
-Copy-Item "${DocGenDir}/assets/logo.svg" -Destination "${DocOutDir}/_site/" -Force
-Copy-Item "${DocGenDir}/assets/toc.yml" -Destination "${DocOutDir}/_site/" -Force
+if ((Get-ChildItem -Path Function: | ? { $_.Name -eq $GetGithubIoDocIndexFn  }).Count -gt 0)
```

Comment:
We should switch this to use Test-Path similar to what Dozie did recently with https://github.com/Azure/azure-sdk-tools/pull/1153/files#diff-f573da0961a88a76bda19315c201497d2095a2b703d876a171120f4202f1c70dR1